src/DataRenderer.tsx

```diff
@@ -202,13 +202,33 @@ function JsonPrimitiveValue({
   );
 }
 
+function DateValue({ field, value, style, lastElement }: JsonRenderProps<Date>) {
+  const stringValue = value.toISOString();
+  const valueStyle = style.otherValue;
+
+  if (field === '') {
+    field = '""';
+  }
+
+  return (
+    <div className={style.basicChildStyle} role='listitem'>
+      {field && <span className={style.label}>{field}:</span>}
+      <span className={valueStyle}>{stringValue}</span>
+      {!lastElement && <span className={style.punctuation}>,</span>}
+    </div>
+  );
+}
+
 export default function DataRender(props: JsonRenderProps<any>) {
   const value = props.value;
   if (DataTypeDetection.isArray(value)) {
     return <JsonArray {...props} />;
   }
 
   if (DataTypeDetection.isObject(value)) {
+    if (DataTypeDetection.isDate(value)) {
+      return <DateValue {...props} />;
```

Comment:
Thanks for the PR!
Even `Date` is not a "primitive" value but still it is rendered in the same way as string/bool/number.
How about to use the same `JsonPrimitiveValue` component, e.g.:

```ts
if (DataTypeDetection.isDate(value)) {
  return <JsonPrimitiveValue {...props} />;
}
```


```ts
} else if (DataTypeDetection.isDate(value)) {
  stringValue = (value as Date).toISOString();
  valueStyle = style.otherValue;
}
```
?

we check for the Date type two times but we don't need to duplicate other logic. Of course "no code duplication" is not the goal, but we have the same pattern - string value and style and checking for the empty key value.

Reply:
updated!